flask_jwt_extended/tokens.py

```diff
@@ -123,21 +123,24 @@ def decode_jwt(encoded_token, secret, algorithm, identity_claim_key,
     :param identity_claim_key: expected key that contains the identity
     :param user_claims_key: expected key that contains the user claims
     :param csrf_value: Expected double submit csrf value
+    :param audience: expected audience in the JWT
     :return: Dictionary containing contents of the JWT
     """
-    # This call verifies the ext, iat, and nbf claims
-    data = jwt.decode(encoded_token, secret, algorithms=[algorithm])
+    # This call verifies the ext, iat, nbf, and aud claims
+    data = jwt.decode(encoded_token, secret, algorithms=[algorithm], audience=audience)
 
     # Make sure that any custom claims we expect in the token are present
     if 'jti' not in data:
-        raise JWTDecodeError("Missing claim: jti")
+        data['jti'] = _create_csrf_token()
```

Comment:
We will need to do something different here, this will cause the same token to receive different jtis on each request, which has implications for things like blacklisting tokens. 

The only other thing I could think to do here would be to set `data['jti'] = None` instead.